src/scan-iterator.ts

```diff
@@ -90,24 +93,18 @@ class ScanIterator<V> implements AsyncIterableIterator<V> {
   }
 
   private async _load(): Promise<void> {
-    if (this._loadPromise) {
-      await this._loadPromise;
-      return;
-    }
-
     if (!this._transaction) {
       this._transaction = await this._getTransaction();
     }
 
-    const responseItems: ScanItem[] = [];
```

Comment:
Hm. I feel like there was a reason for the local but I can't remember what it was.

Reply:
It was needed when we did batching